### ISSUE TITLE
Add names to each test

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#354]
Currently we identify unit tests with a comment at the top:

```ts
  valid: [
    {
      code: stripIndent`
        // with $
        import { Observable, of, EMPTY } from "rxjs";

        const someObservable$ = of(0);
...
```

but the ESLint 8.1 added a `name` property to their rule tester.  We should switch to that instead:

```ts
  valid: [
    {
      name: 'with $',
      code: stripIndent`
        import { Observable, of, EMPTY } from "rxjs";

        const someObservable$ = of(0);
...
```
